### ISSUE TITLE
CA1823 emitted for constant referenced in constructor

### ISSUE THREAD

**Issue by pharring** [dotnet/roslyn-analyzers#1563]
#### Analyzer package

Microsoft.CodeQuality.Analyzers, version [2.6.0](https://www.nuget.org/packages/Microsoft.CodeQuality.Analyzers/2.6.0)

#### Analyzer

CA1823: Avoid unused private fields.

#### Repro steps
Analyze this:
```C#
using System;

namespace Test
{
    public sealed class TestClass
    {
        public TestClass() => TestMethod(c_DefaultValue);

        private static void TestMethod(int value) => Console.WriteLine(value);

        private const int c_DefaultValue = 0x00010000;
    }
}
```

#### Expected behavior
No warnings because `c_DefaultValue` is referenced in the constructor.

#### Actual behavior
warning CA1823: Unused field 'c_DefaultValue'

**Comment by sharwell** [comment]
:+1:

**Comment by mavasani** [comment]
Due to Roslyn bug https://github.com/dotnet/roslyn/issues/26520. I have a PR out to fix it: https://github.com/dotnet/roslyn/pull/26522
